scripts/subctl/getsubctl.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+set -e
+get_operating_system() {
+  case $(uname -s) in
+
+    Darwin*) echo "darwin" ;;
+    Linux*) echo "linux" ;;
+    CYGWIN*) echo "windows" ;;
+    MINGW*) echo "windows" ;;
+    *)
+    error "This installer is only supported on Linux and macOS and Windows. Found $(uname -s)"
+    return 1;;
+  esac
+}
+
+get_architecture() {
+  case $(uname -m) in
+
+    x86_64) echo "amd64" ;;
+    amd64) echo "amd64" ;;
+    i?86) echo "386" ;;
+    *)
+    error "This installer is only for x86_64 and i386 architectures. Found $(uname -m)"
+    return 1;;
+  esac
+}
+
+command_exists() {
+  command -v "$@" > /dev/null 2>&1
+}
+
+download_command() {
+  if command_exists curl; then
+    echo "curl -fsSL"
+  elif command_exists wget; then
+    echo "wget -qO-"
+  else
+    error "curl and wget are missing"
+    return 1
+  fi
+}
+
+GET=$(download_command)
+
+get_subctl_release_url() {
+  version=$1
+  os=$2
+  architecture=$3
+
+  case ${version} in
+    latest) url=https://api.github.com/repos/submariner-io/submariner-operator/releases/latest ;;
+    master) url=https://api.github.com/repos/submariner-io/submariner-operator/releases/tags/latest ;;
+    *) url=https://api.github.com/repos/submariner-io/submariner-operator/releases/tags/${version} ;;
+  esac
+
+  ${GET} "${url}" | grep "browser_download_url.*-${os}-${architecture}" | cut -d\" -f 4
+
+}
+
+finish_cleanup() {
+  if [ "${tmpdir}" != "" ]; then
+    rm -rf "$tmpdir"
+  fi
+}
+
+get_subctl() {
+  version=$1
+  architecture=$2
+  os=$3
+  url=$4
+
+  tmpdir=$(mktemp -d)
+
+  cd "${tmpdir}"
+
+  case ${url} in
+    *tar.xz)
+      ${GET} "${url}" | tar xf -
+      # shellcheck disable=SC2086
+      install_subctl subctl*/subctl*${os}-${architecture}*
+      ;;
+    *) # non tar.xz releases (older)
+      filename=$(basename "${url}")
+      ${GET} "${url}" > "${filename}"
+      install_subctl "${filename}"
+      ;;
+  esac
+}
+
+install_subctl() {
+  bin=$1
+  destdir=~/.local/bin
+  dest=${destdir}/subctl
+
+  mkdir -p "${destdir}"
+  install "${bin}" "${dest}"
+
+  bin_file=$(basename "${bin}")
+  echo "${bin_file} has been installed as ${dest}"
+  echo please update your path with: export PATH=\$PATH:~/.local/bin and consider adding it to ~/.profile
+}
+
+VERSION="${VERSION:-latest}"
+OS=$(get_operating_system)
+ARCHITECTURE=$(get_architecture)
+URL=$(get_subctl_release_url "${VERSION}" "${OS}" "${ARCHITECTURE}")
```

Comment:
```suggestion
URL=$(get_subctl_release_url)
```

Since they're global variables, just use them :)

Reply:
Prff, can we keep the explicit variable passing?, I know bash vars are global by default, but it's a bit harder to follow that way.

I can add locals on the right places to make it explicit. I personally prefer slightly longer but more easy to follow code ':D. But that's opinion, functionally in the end your approach or mine would work...

Reply:
As we discussed on slack, it doesn't add any clarity in this case and would be easier and cleaner to use the globals you already have.